[css-color-4] Add new set of CSS system colors (addresses #3804)

melanierichards opened the pull request:
Preserves language around system colors from css-color-3; moves undeprecated values from the appendix, leaving behind those which are still deprecated.

dbaron commented on css-color-4/Overview.bs:
```diff
@@ -129,7 +129,7 @@ Representing sRGB Colors: the <<color>> type</h2>
 	          <<hwb()>> | <<gray()>> | <<device-cmyk()>> |
 	          <<hex-color>> | <<named-color>> | currentcolor |
-	          <<deprecated-system-color>>
+	          <<system-color>>
```
It looks like you define both `<system-color>` and `<deprecated-system-color>`, so don't both of them still need to be referenced here?  Though I think it would likely be better to have a single term that includes both of them (probably `<system-color>`), but that doesn't look like how it's defined in this PR.

dbaron commented on css-color-4/Overview.bs:
```diff
@@ -2759,11 +2809,11 @@ Default Style Rules</h2>
 
 	Earlier versions of CSS defined several additional named color keywords,
-	the <dfn>&lt;deprecated-system-color></dfn>s,
+	the <dfn id="deprecated-system-color" lt="deprecated system colors" export>system colors</dfn>&lt;deprecated-system-color></dfn>s,
```
This change seems a bit strange; it's not clear why the `<deprecated-system-color>` text remains after the `dfn` element.  It might be OK as it was...

melanierichards replied:
This looks like a weird paste error I didn't catch, thanks @dbaron!

dbaron commented on css-color-4/Overview.bs:
```diff
@@ -2332,6 +2332,56 @@ Converting Between Uncalibrated CMYK and RGB-Based Colors</h3>
 
+
+<h2 id="css-system-colors">
+CSS System Colors</h2>
```
As I mentioned above, I think it would likely be better if something in this section included the deprecated system colors in the `<system-color>` term so that other pieces of spec wouldn't miss half of them.

melanierichards replied:
Besides removing the definition of `<deprecated-system-color>`, would you mind clarifying what you'd like changed here? Move the deprecated list out of the appendix and into this section (though of course still noting they are deprecated)?

svgeesus replied:
@dbaron it looks like this PR is awaiting your clarification of what changes you requested.

melanierichards commented on css-color-4/Overview.bs:
```diff
@@ -2332,6 +2332,56 @@ Converting Between Uncalibrated CMYK and RGB-Based Colors</h3>
+		<dt><dfn>ButtonText</dfn>
+		<dd>Text on push buttons.
+		<dt><dfn>Canvas</dfn>
+		<dd>Window background.
```
I carried forward a lot of language from css-color-3, I like your suggestion better :)

melanierichards commented on css-color-4/Overview.bs:
```diff
@@ -2332,6 +2332,56 @@ Converting Between Uncalibrated CMYK and RGB-Based Colors</h3>
+	
+	Any color property can take one of the following names. Although 
+	these are case-insensitive, it is recommended that the mixed capitalization shown 
+	below be used, to make the names more legible.
```
Carried forward old language here, let me soften based on feedback.

melanierichards commented:
@dbaron @AmeliaBR one thing I wanted to ask after I take the PR out of draft mode was whether it makes sense to have this new section on system colors as its own section or the spec, or if it would make more sense as a subsection under named colors. Do you all have thoughts/opinions on that?

AmeliaBR commented:
> whether it makes sense to have this new section on system colors as its own section of the spec, or if it would make more sense as a subsection under named colors. 

A subsection makes sense. Or maybe [the top-level section](https://drafts.csswg.org/css-color-4/#named-colors) should be renamed "Color keywords", and then have four subsections below it:

- named colors
- system colors
- transparent
- currentcolor

melanierichards commented:
Commit has been updated to include the following changes:

* Moved system colors section under "named colors" section
* Addressed inline feedback, with the exception of clarification on how folks want deprecated system colors integrated into the definition (if other changes made don't suffice)
* Updated language in security & privacy, and referenced from system colors section
* Moved from length-based line breaks to semantic line breaks

AmeliaBR commented:
Thanks for the fast updates, Melanie!

On security & privacy, the issue is not just about spoofing, it's also about fingerprinting. See also #3873.

melanierichards commented:
@AmeliaBR The security & privacy section of the spec already has a mention for the fingerprinting risk, do we feel it needs more detail? Seems like #3873 might be out of scope for this particular PR, or at least we might need to discuss first as a group?

@Crissov @svgeesus was your upvote on Amelia's comment in favor of renaming the "Named colors" section to "Color keywords"? For now I had just moved system colors within the "Named colors" section.

Thanks all for the feedback!

AmeliaBR commented:
I think your current text re security & privacy is accurate. Any discussion of stricter requirements on UAs can happen in the other issue.

Crissov commented:
Iʼm I in the favor of renaming the section and adding sub-headings as suggested.

svgeesus commented:
> @svgeesus was your upvote on Amelia's comment in favor of renaming the "Named colors" section to "Color keywords"?

Yes, it was.

melanierichards commented:
Just renamed "Named colors" section => "Color keywords" as requested (and sheepishly added myself to acknowledgements per @AmeliaBR's comment).

Apologies for the intermittent delay on these last changes!
